packages/mui-material/src/OverridableComponentAugmentation.ts

```diff
@@ -0,0 +1,44 @@
+import * as React from 'react';
+import { DistributiveOmit } from '@mui/types';
+
+declare module '@mui/material/OverridableComponent' {
+  /**
+   * A component whose root component can be controlled via a `component` prop.
+   *
+   * Adjusts valid props based on the type of `component`.
+   */
+  interface OverridableComponent<M extends OverridableTypeMap> {
+    <C extends React.ElementType>(
+      props: {
+        /**
+         * The component used for the root node.
+         * Either a string to use a HTML element or a component.
+         */
+        component: C;
+      } & OverridePropsVer2<M, C>,
+    ): JSX.Element;
+    (props: DefaultComponentPropsVer2<M>): JSX.Element;
+  }
+
+  /**
+   * Props of the component if `component={Component}` is used.
+   */
+  // prettier-ignore
+  type OverridePropsVer2<
+    M extends OverridableTypeMap,
+    C extends React.ElementType,
+  > = (
+    & BaseProps<M>
+    & DistributiveOmit<React.ComponentPropsWithoutRef<C>, keyof BaseProps<M>>
+    & { ref?: React.Ref<any> }
```

Comment:
Have you checked if `ref?: React.Ref<Element>` would work? It would at least give developers some properties of the actual element type.

Reply:
Hmm interesting, I will try that

Reply:
Looks better 👍 At least better than `any` :)